packages/state-transition/src/cache/epochCache.ts

```diff
@@ -505,16 +572,25 @@ export class EpochCache {
       nextEpochTotalActiveBalanceByIncrement: number;
     }
   ): void {
+    this.previousDecisionRoot = this.currentDecisionRoot;
     this.previousShuffling = this.currentShuffling;
-    this.currentShuffling = this.nextShuffling;
-    const currEpoch = this.currentShuffling.epoch;
-    const nextEpoch = currEpoch + 1;
-
-    this.nextShuffling = computeEpochShuffling(
-      state,
-      epochTransitionCache.nextEpochShufflingActiveValidatorIndices,
-      nextEpoch
-    );
+
+    this.currentDecisionRoot = this.nextDecisionRoot;
+    if (this.nextShuffling) {
+      // was already pulled by the api or another method on EpochCache
+      this.currentShuffling = this.nextShuffling;
+    } else {
```

Comment:
would be nice to track this in metric here

Reply:
Updated